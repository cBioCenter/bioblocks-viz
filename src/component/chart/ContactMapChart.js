"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var React = require("react");
var component_1 = require("~bioblocks-viz~/component");
var helper_1 = require("~bioblocks-viz~/helper");
exports.generateChartDataEntry = function (hoverinfo, color, name, subtitle, nodeSize, points, extra) {
    if (extra === void 0) { extra = {}; }
    return (tslib_1.__assign({ hoverinfo: hoverinfo, marker: typeof color === 'string'
            ? { color: new Array(points.length * 2).fill(color) }
            : {
                colorscale: [[0, color.start], [1, color.end]],
            }, mode: 'lines+markers', name: name,
        nodeSize: nodeSize,
        points: points,
        subtitle: subtitle }, extra));
};
/**
 * Intermediary between a ContactMap and a PlotlyChart.
 *
 * Will transform data and setup layout from science/bioblocks into a format suitable for Plotly consumption.
 * @extends {React.Component<IContactMapChartProps, any>}
 */
var ContactMapChart = /** @class */ (function (_super) {
    tslib_1.__extends(ContactMapChart, _super);
    function ContactMapChart(props) {
        var _this = _super.call(this, props) || this;
        _this.toggleLegendVisibility = function () {
            _this.setState({
                showlegend: !_this.state.showlegend,
            });
        };
        _this.state = {
            numLegends: 0,
            plotlyData: [],
            showlegend: false,
        };
        return _this;
    }
    ContactMapChart.prototype.componentDidMount = function () {
        this.setupData();
    };
    ContactMapChart.prototype.componentDidUpdate = function (prevProps) {
        var _a = this.props, contactData = _a.contactData, secondaryStructures = _a.secondaryStructures, selectedSecondaryStructures = _a.selectedSecondaryStructures;
        if (prevProps.contactData !== contactData ||
            prevProps.secondaryStructures !== secondaryStructures ||
            prevProps.selectedSecondaryStructures !== selectedSecondaryStructures) {
            this.setupData();
        }
    };
    ContactMapChart.prototype.render = function () {
        var _a = this.props, configurations = _a.configurations, contactData = _a.contactData, isDataLoading = _a.isDataLoading, legendModifiers = _a.legendModifiers, marginModifiers = _a.marginModifiers, range = _a.range, showConfigurations = _a.showConfigurations, passThroughProps = tslib_1.__rest(_a, ["configurations", "contactData", "isDataLoading", "legendModifiers", "marginModifiers", "range", "showConfigurations"]);
        var _b = this.state, plotlyData = _b.plotlyData, showlegend = _b.showlegend;
        return (React.createElement(component_1.PlotlyChart, tslib_1.__assign({ data: plotlyData, showLoader: isDataLoading, layout: {
                legend: {
                    orientation: 'h',
                    y: legendModifiers.y,
                    yanchor: 'bottom',
                },
                margin: {
                    b: marginModifiers.b,
                    l: marginModifiers.l,
                },
                showlegend: showlegend,
                xaxis: {
                    fixedrange: true,
                    nticks: 10,
                    range: [0, range],
                    rangemode: 'nonnegative',
                    showline: true,
                    tickmode: 'auto',
                    title: 'Residue #',
                },
                yaxis: {
                    fixedrange: true,
                    nticks: 10,
                    range: [range, 0],
                    rangemode: 'nonnegative',
                    showline: true,
                    tickmode: 'auto',
                    title: 'Residue #',
                },
            } }, passThroughProps)));
    };
    /**
     * Sets up the chart and axis data for the ContactMap.
     *
     * Transforms all data from bioblocks terminology to data properly formatted for Plotly consumption.
     */
    ContactMapChart.prototype.setupData = function () {
        var _a = this.props, contactData = _a.contactData, dataTransformFn = _a.dataTransformFn, secondaryStructures = _a.secondaryStructures, secondaryStructureColors = _a.secondaryStructureColors, selectedSecondaryStructures = _a.selectedSecondaryStructures;
        var plotlyData = tslib_1.__spread(contactData.map(function (entry) { return dataTransformFn(entry, true); }));
        secondaryStructures.forEach(function (secondaryStructure, index) {
            var axis = new component_1.SecondaryStructureAxis(secondaryStructure, 3, index + 2, secondaryStructureColors);
            plotlyData.push.apply(plotlyData, tslib_1.__spread(axis.xAxes, axis.yAxes));
        });
        var highlightedAxes = new Array();
        selectedSecondaryStructures.forEach(function (selectedStructure, index) {
            var axis = new component_1.AuxiliaryAxis(selectedStructure, index + 2, new helper_1.ColorMapper(new Map(), 'orange'));
            highlightedAxes.push.apply(highlightedAxes, tslib_1.__spread(axis.highlightedXAxes, axis.highlightedYAxes));
        });
        this.setState({
            numLegends: new Set(plotlyData.filter(function (datum) { return datum.showlegend !== false && datum.name !== undefined; }).map(function (legend) { return legend.name; })).size,
            // Makes sure that highlighted axis is behind the axis.
            plotlyData: tslib_1.__spread(highlightedAxes, plotlyData),
        });
    };
    ContactMapChart.defaultProps = {
        configurations: new Array(),
        dataTransformFn: helper_1.generateScatterGLData,
        height: '100%',
        isDataLoading: false,
        legendModifiers: {
            y: -0.4,
        },
        marginModifiers: {
            b: 67,
            l: 67,
        },
        range: 33000,
        secondaryStructures: [],
        selectedSecondaryStructures: [],
        selectedSecondaryStructuresColor: '#feb83f',
        showConfigurations: true,
        width: '100%',
    };
    return ContactMapChart;
}(React.Component));
exports.ContactMapChart = ContactMapChart;
//# sourceMappingURL=ContactMapChart.js.map