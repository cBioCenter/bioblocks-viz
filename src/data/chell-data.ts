// Type definitions for Chell Visualization Library
// Project: https://github.com/cBioCenter/chell-viz
// Definitions by: cBioCenter @ Dana-Farber <https://github.com/cBioCenter/>, James Lindsay <https://github.com/jim-bo>, Nick Gauthier <https://github.com/npgauth>, Drew Diamantoukos <https://github.com/MercifulCode>,
// TypeScript Version: 2.8
import { Structure } from 'ngl';
import { ISpringGraphData } from 'spring';
import Chell1DSection from './Chell1DSection';
import { ChellPDB } from './ChellPDB';
import { CouplingContainer } from './CouplingContainer';

export type CONTACT_MAP_DATA_TYPE = IContactMapData;
export type NGL_DATA_TYPE = Structure;
export type SPRING_DATA_TYPE = ISpringGraphData;
export type T_SNE_DATA_TYPE = number[][];

export type CHELL_DATA_TYPE = CONTACT_MAP_DATA_TYPE | NGL_DATA_TYPE | SPRING_DATA_TYPE | T_SNE_DATA_TYPE;

// TODO Better define Cell type to not be indices.
/** Shared data between Spring/T-SNE. Refers to index in points collection. */
export type CELL_TYPE = number;
export type RESIDUE_TYPE = number;

export enum VIZ_TYPE {
  CONTACT_MAP = 'Contact Map',
  NGL = 'NGL',
  SPRING = 'Spring',
  'T-SNE' = 'T-SNE',
}

export interface IContactMapData {
  couplingScores: CouplingContainer;
  pdbData?: ChellPDB;
  secondaryStructures: ISecondaryStructureData[];
}

export interface IMonomerContact {
  i: number;
  j: number;
  dist: number;
}

export interface ICouplingScore {
  i: number;
  A_i?: string;
  j: number;
  A_j?: string;
  fn?: number;
  cn?: number;
  segment_i?: string;
  segment_j?: string;
  probability?: number;
  dist_intra?: number;
  dist_multimer?: number;
  dist: number;
  precision?: number;
}

export enum CONTACT_DISTANCE_PROXIMITY {
  CLOSEST = 'CLOSEST',
  C_ALPHA = 'C-Î±',
}

export enum CONFIGURATION_COMPONENT_TYPE {
  RADIO = 'RADIO',
  SLIDER = 'SLIDER',
}

export interface ISecondaryStructureData {
  resno: number;
  structId: SECONDARY_STRUCTURE_KEYS;
}

export type SECONDARY_STRUCTURE_SECTION = Chell1DSection<SECONDARY_STRUCTURE_KEYS>;
export type SECONDARY_STRUCTURE = Array<Chell1DSection<SECONDARY_STRUCTURE_KEYS>>;

export interface IResiduePair {
  i: number;
  j: number;
}

export enum SECONDARY_STRUCTURE_CODES {
  'G' = '310_HELIX',
  'H' = 'ALPHA_HELIX',
  'I' = 'PI_HELIX',
  'T' = 'HYDROGEN_BONDED_TURN',
  'E' = 'BETA_SHEET',
  'B' = 'BETA_BRIDGE',
  'S' = 'BEND',
  'C' = 'COIL',
}

export type SECONDARY_STRUCTURE_KEYS = keyof typeof SECONDARY_STRUCTURE_CODES;

export interface ISecondaryStructure {
  code: SECONDARY_STRUCTURE_CODES;
  name: string;
  minResidueLength: number;
}
