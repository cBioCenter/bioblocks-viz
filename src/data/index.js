"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
tslib_1.__exportStar(require("./AminoAcid"), exports);
tslib_1.__exportStar(require("./Nucleotide"), exports);
tslib_1.__exportStar(require("./AnatomogramMapping"), exports);
tslib_1.__exportStar(require("./Bioblocks1DSection"), exports);
tslib_1.__exportStar(require("./BioblocksConfig"), exports);
tslib_1.__exportStar(require("./BioblocksData"), exports);
tslib_1.__exportStar(require("./BioblocksPDB"), exports);
tslib_1.__exportStar(require("./CodonTable"), exports);
tslib_1.__exportStar(require("./CouplingContainer"), exports);
tslib_1.__exportStar(require("./Dataset"), exports);
tslib_1.__exportStar(require("./IUPACData"), exports);
tslib_1.__exportStar(require("./Markers"), exports);
tslib_1.__exportStar(require("./PfamProtein"), exports);
tslib_1.__exportStar(require("./PlotlyData"), exports);
tslib_1.__exportStar(require("./Protein"), exports);
tslib_1.__exportStar(require("./Seq"), exports);
tslib_1.__exportStar(require("./SeqIO"), exports);
tslib_1.__exportStar(require("./SeqRecord"), exports);
tslib_1.__exportStar(require("./TintedBioblocks1DSection"), exports);
tslib_1.__exportStar(require("./event"), exports);
//# sourceMappingURL=index.js.map