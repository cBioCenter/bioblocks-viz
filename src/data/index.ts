export * from './AminoAcid';
export * from './Chell1DSection';
export * from './ChellConfig';
export * from './ChellData';
export * from './ChellPDB';
export * from './CouplingContainer';
export * from './PlotlyData';
export * from './Protein';
export * from './Spring';
export * from './TintedChell1DSection';

export * from './event';
