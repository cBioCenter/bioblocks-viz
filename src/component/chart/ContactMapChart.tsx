import * as plotly from 'plotly.js-gl2d-dist';
import * as React from 'react';

import { RESIDUE_TYPE, SECONDARY_STRUCTURE } from '../../data/chell-data';

import { ChellWidgetConfig } from '../../data/ChellConfig';
import { generateScatterGLData } from '../../helper/PlotlyHelper';
import { SettingsPanel } from '../widget/SettingsPanel';
import { AuxiliaryAxis } from './AuxiliaryAxis';
import { defaultPlotlyLayout, IPlotlyData, PlotlyChart } from './PlotlyChart';
import { SecondaryStructureAxis } from './SecondaryStructureAxis';

export interface IContactMapChartProps {
  candidateResidues: RESIDUE_TYPE[];
  configurations: ChellWidgetConfig[];
  contactData: IContactMapChartData[];
  dataTransformFn: (entry: IContactMapChartData, mirrorPoints: boolean) => Partial<IPlotlyData>;
  heightModifier: number;
  legendModifiers: {
    y: number;
  };
  marginModifiers: {
    b: number;
    l: number;
  };
  onClickCallback?: (...args: any[]) => void;
  onHoverCallback?: (...args: any[]) => void;
  onSelectedCallback?: (...args: any[]) => void;
  onUnHoverCallback?: (...args: any[]) => void;
  range: number;
  secondaryStructures: SECONDARY_STRUCTURE[];
  selectedSecondaryStructures: SECONDARY_STRUCTURE[];
}

export interface IContactMapChartState {
  plotlyData: Array<Partial<IPlotlyData>>;
}

export interface IContactMapChartData extends Partial<IPlotlyData> {
  name: string;
  nodeSize: number;
  points: IContactMapChartPoint[];
  subtitle?: string;
}

export const generateChartDataEntry = (
  hoverinfo: plotly.ScatterData['hoverinfo'],
  color: string | { start: string; end: string },
  name: string,
  subtitle: string,
  nodeSize: number,
  points: IContactMapChartPoint[],
  extra: Partial<IPlotlyData> = {},
): IContactMapChartData => ({
  hoverinfo,
  marker:
    typeof color === 'string'
      ? { color: new Array(points.length * 2).fill(color) }
      : {
          colorscale: [
            [0.0, 'rgb(12,50,102)'],
            [0.1, 'rgb(17,83,150)'],
            [0.2, 'rgb(40,114,175)'],
            [0.3, 'rgb(71,147,193)'],
            [0.4, 'rgb(111,175,209)'],
            [0.5, 'rgb(160,202,222)'],
            [0.6, 'rgb(200,219,237)'],
            [0.7, 'rgb(224,235,246)'],
            [0.8, 'rgb(247,251,255)'],
            [0.9, 'rgb(249,253,255)'],
            [1.0, 'rgb(255,255,255)'],
          ],
        },
  mode: 'lines+markers',
  name,
  nodeSize,
  points,
  subtitle,
  ...extra,
});

export interface IContactMapChartPoint {
  dist: number;
  i: number;
  j: number;
}

/**
 * Intermediary between a ContactMap and a PlotlyChart.
 *
 * Will transform data and setup layout from science/chell data type into the Plotly type.
 * @extends {React.Component<IContactMapChartProps, any>}
 */
class ContactMapChart extends React.Component<IContactMapChartProps, IContactMapChartState> {
  public static defaultProps = {
    candidateResidues: new Array<RESIDUE_TYPE>(),
    configurations: new Array<ChellWidgetConfig>(),
    dataTransformFn: generateScatterGLData,
    heightModifier: 0.3,
    legendModifiers: {
      y: -0.4,
    },
    marginModifiers: {
      b: 75,
      l: 75,
    },
    range: 100,
    secondaryStructures: [],
    selectedSecondaryStructures: [],
  };

  constructor(props: IContactMapChartProps) {
    super(props);
    this.state = {
      plotlyData: [],
    };
  }

  public componentDidMount() {
    this.setupData();
  }

  public componentDidUpdate(prevProps: IContactMapChartProps) {
    const { contactData, secondaryStructures, selectedSecondaryStructures } = this.props;
    if (
      prevProps.contactData !== contactData ||
      prevProps.secondaryStructures !== secondaryStructures ||
      prevProps.selectedSecondaryStructures !== selectedSecondaryStructures
    ) {
      this.setupData();
    }
  }

  public render() {
    const {
      configurations,
      contactData,
      heightModifier,
      legendModifiers,
      marginModifiers,
      range,
      ...props
    } = this.props;
    const { plotlyData } = this.state;

    return (
      <SettingsPanel configurations={configurations}>
        <PlotlyChart
          data={plotlyData}
          layout={{
            height: defaultPlotlyLayout.height! + defaultPlotlyLayout.height! * heightModifier,
            legend: {
              orientation: 'h',
              y: legendModifiers.y,
              yanchor: 'bottom',
            },
            margin: {
              b: marginModifiers.b,
              l: marginModifiers.l,
            },
            showlegend: true,
            xaxis: {
              autorange: false,
              nticks: 10,
              range: [1, range],
              showline: true,
              tickmode: 'auto',
              title: 'Residue #',
            },
            yaxis: {
              autorange: false,
              nticks: 10,
              range: [1, range].reverse(),
              showline: true,
              tickmode: 'auto',
              title: 'Residue #',
            },
          }}
          {...props}
        />
      </SettingsPanel>
    );
  }

  /**
   * Sets up the chart and axis data for the ContactMap.
   *
   * Transforms all data from chell terminology to data properly formatted for Plotly consumption.
   */
  protected setupData() {
    const { contactData, dataTransformFn, secondaryStructures, selectedSecondaryStructures } = this.props;
    const plotlyData = [...contactData.map(entry => dataTransformFn(entry, true))];
    secondaryStructures.forEach((secondaryStructure, index) => {
      const axis = new SecondaryStructureAxis(secondaryStructure, 5, index + 2);
      plotlyData.push(...axis.xAxes, ...axis.yAxes);
    });

    selectedSecondaryStructures.forEach((selectedStructure, index) => {
      const axis = new AuxiliaryAxis(selectedStructure, index + 2, 'orange');
      plotlyData.push(...axis.xAxes, ...axis.yAxes);
    });

    this.setState({
      plotlyData,
    });
  }
}

export default ContactMapChart;
