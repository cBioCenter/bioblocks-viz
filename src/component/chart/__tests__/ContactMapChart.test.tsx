import { shallow } from 'enzyme';
import * as React from 'react';

import { SECONDARY_STRUCTURE, SECONDARY_STRUCTURE_CODES, SECONDARY_STRUCTURE_KEYS } from '../../../data/chell-data';
import Chell1DSection from '../../../data/Chell1DSection';
import ContactMapChart from '../ContactMapChart';

describe('ContactMapChart', () => {
  const genSeqEntry = (
    structId: keyof typeof SECONDARY_STRUCTURE_CODES,
    resno: number,
    length: number = 1,
  ): SECONDARY_STRUCTURE => [new Chell1DSection<SECONDARY_STRUCTURE_KEYS>(structId, resno, resno + length - 1)];

  const emptyData = [
    {
      color: '',
      name: '',
      nodeSize: 0,
      points: [],
    },
  ];

  it('Should match existing snapshot when given simple data.', () => {
    const wrapper = shallow(<ContactMapChart contactData={emptyData} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match existing snapshot when given data with extra axes.', () => {
    const wrapper = shallow(
      <ContactMapChart
        contactData={emptyData}
        secondaryStructures={[[...genSeqEntry('H', 0, 3), ...genSeqEntry('E', 3, 1)]]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match existing snapshot when given data with multiple secondary structures.', () => {
    const wrapper = shallow(
      <ContactMapChart
        contactData={emptyData}
        secondaryStructures={[
          [...genSeqEntry('H', 0, 3), ...genSeqEntry('E', 3, 1)],
          [...genSeqEntry('C', 0, 3), ...genSeqEntry('H', 11, 4)],
        ]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
