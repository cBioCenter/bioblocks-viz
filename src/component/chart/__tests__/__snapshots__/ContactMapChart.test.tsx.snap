// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMapChart Should match existing snapshot when given data with extra axes. 1`] = `
<SettingsPanel
  configurations={Array []}
  direction="left"
  inverted={true}
  showConfigurations={true}
  width="100%"
>
  <PlotlyChart
    candidateResidues={Array []}
    config={Object {}}
    data={
      Array [
        Object {
          "hoverinfo": "x+y+z",
          "marker": Object {
            "color": Array [],
            "size": 0,
          },
          "mode": "markers",
          "name": " undefined",
          "text": Array [],
          "type": "scattergl",
          "x": Array [],
          "y": Array [],
          "z": Array [],
        },
        Object {
          "connectgaps": false,
          "hoverinfo": "none",
          "line": Object {
            "color": "blue",
            "shape": "spline",
            "smoothing": 1.3,
            "width": 1.5,
          },
          "marker": Object {
            "symbol": Array [],
          },
          "mode": "lines",
          "name": "H",
          "orientation": "h",
          "showlegend": false,
          "type": "scatter",
          "x": Array [
            0,
            0,
            1,
            2,
            2,
          ],
          "xaxis": "x",
          "y": Array [
            null,
            0,
            0.8414709848078965,
            0.9092974268256817,
            null,
          ],
          "yaxis": "y2",
        },
        Object {
          "connectgaps": false,
          "hoverinfo": "none",
          "line": Object {
            "color": "blue",
            "shape": "spline",
            "smoothing": 1.3,
            "width": 1.5,
          },
          "marker": Object {
            "symbol": Array [],
          },
          "mode": "lines",
          "name": "H",
          "orientation": "v",
          "showlegend": false,
          "type": "scatter",
          "x": Array [
            null,
            0,
            0.8414709848078965,
            0.9092974268256817,
            null,
          ],
          "xaxis": "x2",
          "y": Array [
            0,
            0,
            1,
            2,
            2,
          ],
          "yaxis": "y",
        },
      ]
    }
    dataTransformFn={[Function]}
    layout={
      Object {
        "height": 400,
        "legend": Object {
          "orientation": "h",
          "y": -0.4,
          "yanchor": "bottom",
        },
        "margin": Object {
          "b": 65,
          "l": 65,
        },
        "showlegend": true,
        "width": 400,
        "xaxis": Object {
          "autorange": false,
          "nticks": 10,
          "range": Array [
            1,
            100,
          ],
          "showline": true,
          "tickmode": "auto",
          "title": "Residue #",
        },
        "yaxis": Object {
          "autorange": false,
          "nticks": 10,
          "range": Array [
            100,
            1,
          ],
          "showline": true,
          "tickmode": "auto",
          "title": "Residue #",
        },
      }
    }
    secondaryStructures={
      Array [
        Array [
          Chell1DSection {
            "label": "H",
            "sectionEnd": 2,
            "sectionStart": 0,
          },
          Chell1DSection {
            "label": "E",
            "sectionEnd": 3,
            "sectionStart": 3,
          },
        ],
      ]
    }
    selectedSecondaryStructures={Array []}
    selectedSecondaryStructuresColor="#feb83f"
    showLoader={true}
  />
</SettingsPanel>
`;

exports[`ContactMapChart Should match existing snapshot when given data with multiple secondary structures. 1`] = `
<SettingsPanel
  configurations={Array []}
  direction="left"
  inverted={true}
  showConfigurations={true}
  width="100%"
>
  <PlotlyChart
    candidateResidues={Array []}
    config={Object {}}
    data={
      Array [
        Object {
          "hoverinfo": "x+y+z",
          "marker": Object {
            "color": Array [],
            "size": 0,
          },
          "mode": "markers",
          "name": " undefined",
          "text": Array [],
          "type": "scattergl",
          "x": Array [],
          "y": Array [],
          "z": Array [],
        },
        Object {
          "connectgaps": false,
          "hoverinfo": "none",
          "line": Object {
            "color": "blue",
            "shape": "spline",
            "smoothing": 1.3,
            "width": 1.5,
          },
          "marker": Object {
            "symbol": Array [],
          },
          "mode": "lines",
          "name": "H",
          "orientation": "h",
          "showlegend": false,
          "type": "scatter",
          "x": Array [
            0,
            0,
            1,
            2,
            2,
          ],
          "xaxis": "x",
          "y": Array [
            null,
            0,
            0.8414709848078965,
            0.9092974268256817,
            null,
          ],
          "yaxis": "y2",
        },
        Object {
          "connectgaps": false,
          "hoverinfo": "none",
          "line": Object {
            "color": "blue",
            "shape": "spline",
            "smoothing": 1.3,
            "width": 1.5,
          },
          "marker": Object {
            "symbol": Array [],
          },
          "mode": "lines",
          "name": "H",
          "orientation": "v",
          "showlegend": false,
          "type": "scatter",
          "x": Array [
            null,
            0,
            0.8414709848078965,
            0.9092974268256817,
            null,
          ],
          "xaxis": "x2",
          "y": Array [
            0,
            0,
            1,
            2,
            2,
          ],
          "yaxis": "y",
        },
        Object {
          "connectgaps": false,
          "hoverinfo": "none",
          "line": Object {
            "color": "red",
            "shape": "spline",
            "smoothing": 1.3,
            "width": 1.5,
          },
          "marker": Object {
            "symbol": Array [],
          },
          "mode": "lines",
          "name": "C",
          "orientation": "h",
          "showlegend": false,
          "type": "scatter",
          "x": Array [
            0,
            0,
            1,
            2,
            2,
          ],
          "xaxis": "x",
          "y": Array [
            null,
            0,
            0,
            0,
            null,
          ],
          "yaxis": "y3",
        },
        Object {
          "connectgaps": false,
          "hoverinfo": "none",
          "line": Object {
            "color": "blue",
            "shape": "spline",
            "smoothing": 1.3,
            "width": 1.5,
          },
          "marker": Object {
            "symbol": Array [],
          },
          "mode": "lines",
          "name": "H",
          "orientation": "h",
          "showlegend": false,
          "type": "scatter",
          "x": Array [
            11,
            11,
            12,
            13,
            14,
            14,
          ],
          "xaxis": "x",
          "y": Array [
            null,
            -0.9999902065507035,
            -0.5365729180004349,
            0.4201670368266409,
            0.9906073556948704,
            null,
          ],
          "yaxis": "y3",
        },
        Object {
          "connectgaps": false,
          "hoverinfo": "none",
          "line": Object {
            "color": "red",
            "shape": "spline",
            "smoothing": 1.3,
            "width": 1.5,
          },
          "marker": Object {
            "symbol": Array [],
          },
          "mode": "lines",
          "name": "C",
          "orientation": "v",
          "showlegend": false,
          "type": "scatter",
          "x": Array [
            null,
            0,
            0,
            0,
            null,
          ],
          "xaxis": "x3",
          "y": Array [
            0,
            0,
            1,
            2,
            2,
          ],
          "yaxis": "y",
        },
        Object {
          "connectgaps": false,
          "hoverinfo": "none",
          "line": Object {
            "color": "blue",
            "shape": "spline",
            "smoothing": 1.3,
            "width": 1.5,
          },
          "marker": Object {
            "symbol": Array [],
          },
          "mode": "lines",
          "name": "H",
          "orientation": "v",
          "showlegend": false,
          "type": "scatter",
          "x": Array [
            null,
            -0.9999902065507035,
            -0.5365729180004349,
            0.4201670368266409,
            0.9906073556948704,
            null,
          ],
          "xaxis": "x3",
          "y": Array [
            11,
            11,
            12,
            13,
            14,
            14,
          ],
          "yaxis": "y",
        },
      ]
    }
    dataTransformFn={[Function]}
    layout={
      Object {
        "height": 400,
        "legend": Object {
          "orientation": "h",
          "y": -0.4,
          "yanchor": "bottom",
        },
        "margin": Object {
          "b": 65,
          "l": 65,
        },
        "showlegend": true,
        "width": 400,
        "xaxis": Object {
          "autorange": false,
          "nticks": 10,
          "range": Array [
            1,
            100,
          ],
          "showline": true,
          "tickmode": "auto",
          "title": "Residue #",
        },
        "yaxis": Object {
          "autorange": false,
          "nticks": 10,
          "range": Array [
            100,
            1,
          ],
          "showline": true,
          "tickmode": "auto",
          "title": "Residue #",
        },
      }
    }
    secondaryStructures={
      Array [
        Array [
          Chell1DSection {
            "label": "H",
            "sectionEnd": 2,
            "sectionStart": 0,
          },
          Chell1DSection {
            "label": "E",
            "sectionEnd": 3,
            "sectionStart": 3,
          },
        ],
        Array [
          Chell1DSection {
            "label": "C",
            "sectionEnd": 2,
            "sectionStart": 0,
          },
          Chell1DSection {
            "label": "H",
            "sectionEnd": 14,
            "sectionStart": 11,
          },
        ],
      ]
    }
    selectedSecondaryStructures={Array []}
    selectedSecondaryStructuresColor="#feb83f"
    showLoader={true}
  />
</SettingsPanel>
`;

exports[`ContactMapChart Should match existing snapshot when given simple data. 1`] = `
<SettingsPanel
  configurations={Array []}
  direction="left"
  inverted={true}
  showConfigurations={true}
  width="100%"
>
  <PlotlyChart
    candidateResidues={Array []}
    config={Object {}}
    data={
      Array [
        Object {
          "hoverinfo": "x+y+z",
          "marker": Object {
            "color": Array [],
            "size": 0,
          },
          "mode": "markers",
          "name": " undefined",
          "text": Array [],
          "type": "scattergl",
          "x": Array [],
          "y": Array [],
          "z": Array [],
        },
      ]
    }
    dataTransformFn={[Function]}
    layout={
      Object {
        "height": 400,
        "legend": Object {
          "orientation": "h",
          "y": -0.4,
          "yanchor": "bottom",
        },
        "margin": Object {
          "b": 65,
          "l": 65,
        },
        "showlegend": true,
        "width": 400,
        "xaxis": Object {
          "autorange": false,
          "nticks": 10,
          "range": Array [
            1,
            100,
          ],
          "showline": true,
          "tickmode": "auto",
          "title": "Residue #",
        },
        "yaxis": Object {
          "autorange": false,
          "nticks": 10,
          "range": Array [
            100,
            1,
          ],
          "showline": true,
          "tickmode": "auto",
          "title": "Residue #",
        },
      }
    }
    secondaryStructures={Array []}
    selectedSecondaryStructures={Array []}
    selectedSecondaryStructuresColor="#feb83f"
    showLoader={true}
  />
</SettingsPanel>
`;
